opendse-io/src/main/java/net/sf/opendse/io/SpecificationWriterAnonymized.java

```diff
@@ -0,0 +1,282 @@
+/*******************************************************************************
+ * Copyright (c) 2015 OpenDSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *******************************************************************************/
+package net.sf.opendse.io;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.collections15.Transformer;
+
+import edu.uci.ics.jung.graph.util.EdgeType;
+import edu.uci.ics.jung.graph.util.Pair;
+import net.sf.opendse.model.Application;
+import net.sf.opendse.model.Architecture;
+import net.sf.opendse.model.Dependency;
+import net.sf.opendse.model.Link;
+import net.sf.opendse.model.Mapping;
+import net.sf.opendse.model.Mappings;
+import net.sf.opendse.model.Resource;
+import net.sf.opendse.model.Routings;
+import net.sf.opendse.model.Specification;
+import net.sf.opendse.model.Task;
+
+/**
+ * The {@code SpecificationWriter} writes an anonymized {@code Specification} to
+ * a {@code File}. Anonymization includes {@code Task}, {@code Dependency},
+ * {@code Resource}, {@code Link} and {@code Mapping} ids.

```

Comment:
Maybe I would put it that way: The anonymization covers the IDs of {@code Task}, {@code Dependency}, {@code Resource}, {@code Link} and {@code Mapping}. Note that anonymization keeps all other {@code Attributes} as given in the {@code Specification}.